scripts/portfolioView.js

```diff
@@ -0,0 +1,91 @@
+(function(module){
+  var portfolioView = {};
+  var $headerPosition = $('header').offset().top;
+
+  // helper functions
+  // toggle sticky header class based on scroll position
+  function stickyHeader(){
+    if($(this).scrollTop() > $headerPosition){
+      $('header').addClass('sticky');
+    } else {
+      $('header').removeClass('sticky');
+    }
+  };
+
+  // on click build detailed view of project
+  portfolioView.detailHandler = function (){
+    $('#portfolio').on('click','.project-preview',function(){
+      $('.project-details').insertAfter($(this));
+      var $clickedItem = $(this).data('project');
+      PortfolioItem.all.forEach(function(a){
+        if(a.project === $clickedItem){
+          $('.project-details').html(a.updateDetailModal());
+          setTimeout(function(){
+            $('.project-details').addClass('expanded');
+          }, 100);
+        }
+      });
+    });
+  };
+
+  // on click scroll to matching section
+  portfolioView.navHandler = function(){
+    $('header').on('click','*[data-link]',function(e){
+      e.preventDefault();
+      var $scrollTo = $(this).data('link');
+      $target = $('section[data-content="' + $scrollTo + '"]');
+      $('html, body').animate({
+        scrollTop: $target.offset().top - 50
+      }, 500);
+    });
+  };
+
+  // on click close detail view
+  portfolioView.closeHandler = function(){
+    $('.project-details').on('click','.close',function(e){
+      e.preventDefault();
+      $('.project-details').removeClass('expanded');
+    });
+  };
+
+  // toggle mobile menu
+  portfolioView.toggleMobileMenu = function(){
+    $('.mobile-menu').on('click',function(e){
+      e.preventDefault();
+      $('nav ul').slideToggle();
```

Comment:
BUG: if you are in mobile view and you hide the menu then expand your browser to desktop the nav will be hidden. Resolve through CSS or JS.